Test failure on 32-bit platforms: test_resample_img_segmentation_fault

<!--Provide a brief description of the bug.-->
On 32-bit platforms, `test_resample_img_segmentation_fault` fails:

```
=================================== FAILURES ===================================
_____________________ test_resample_img_segmentation_fault _____________________
    @pytest.mark.skipif(os.environ.get('APPVEYOR') == 'True',
                        reason='This test too slow (7-8 minutes) on AppVeyor')
    @pytest.mark.skipif((os.environ.get('TRAVIS') == 'true'
                         and os.environ.get('TRAVIS_CPU_ARCH') == 'arm64'),
                        reason='This test does not run on ARM arch.')
    def test_resample_img_segmentation_fault():
        # see https://github.com/nilearn/nilearn/issues/346
        shape_in = (64, 64, 64)
        aff_in = np.diag([2., 2., 2., 1.])
        aff_out = np.diag([3., 3., 3., 1.])
        # fourth_dim = 1024 works fine but for 1025 creates a segmentation
        # fault with scipy < 0.14.1
        fourth_dim = 1025

        try:
>           data = np.ones(shape_in + (fourth_dim, ), dtype=np.float64)
nilearn/image/tests/test_resampling.py:808:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
shape = (64, 64, 64, 1025), dtype = <class 'numpy.float64'>, order = 'C'
    @set_array_function_like_doc
    @set_module('numpy')
    def ones(shape, dtype=None, order='C', *, like=None):
        """
        Return a new array of given shape and type, filled with ones.
[… omitted further docstring content …]
        """
        if like is not None:
            return _ones_with_like(shape, dtype=dtype, order=order, like=like)

>       a = empty(shape, dtype, order)
E       ValueError: array is too big; `arr.size * arr.dtype.itemsize` is larger than the maximum possible size.
/usr/lib/python3.10/site-packages/numpy/core/numeric.py:204: ValueError
```

<!--Please fill in the following information, to the best of your ability.-->
Nilearn version:

### Expected behavior

Test passes.

### Actual behavior

Test fails; details above.

### Steps and code to reproduce bug

Run the test suite on a 32-bit platform (including a 32-bit x86 chroot on an x86_64 machine).

The test attempts to create an array of size (64×64×64×1025 values)×8-bytes/value = 2149580800, which exceeds the maximum size representable in a signed word on a 32-bit platform (`0x7fffffff` = 2147483647 = `2**31-1`).

The test should be skipped on 32-bit platforms.

--- a/nilearn/_utils/testing.py
+++ b/nilearn/_utils/testing.py
@@ -48,6 +48,11 @@ except ImportError:
     memory_usage = memory_used = None
 
 
+def is_64bit() -> bool:
+    """Returns True if python is run on 64bits."""
+    return sys.maxsize > 2**32
+
+
 def check_deprecation(func, match=None):
     @functools.wraps(func)
     def wrapped(*args, **kwargs):

--- a/nilearn/image/tests/test_resampling.py
+++ b/nilearn/image/tests/test_resampling.py
@@ -790,6 +790,8 @@ def test_coord_transform_trivial():
     assert x.shape == x_.shape
 
 
+@pytest.mark.skipif(not testing.is_64bit(),
+                    reason='This test only runs on 64bits machines.')
 @pytest.mark.skipif(os.environ.get('APPVEYOR') == 'True',
                     reason='This test too slow (7-8 minutes) on AppVeyor')
 @pytest.mark.skipif((os.environ.get('TRAVIS') == 'true'

